Adding blobstore package which supports Swift

dhlambert opened the pull request:
Extension to GWC which adds native support for Swift. This allows OpenStack Cloud deployments to easily use GeoWebCache without the support from external libraries such as S3Proxy. This has been implemented using the open source multi-cloud toolkit jclouds (https://jclouds.apache.org/).

**Additions:**
- Swift Blobstore package containing blobstore class and supporting config provider classes
- Additions to configuration and pom files for swift
- Unit tests covering blobstore class

dhlambert commented:
Seems the the output of the failing Travis tests produce log output at the debug level when building the web module. Have had a look into this but still not sure why this is occurring and what changes might have caused this?

jodygarnett commented on geowebcache/swiftblob/src/main/java/org/geowebcache/swift/SwiftBlobStoreConfigProvider.java:
```diff
@@ -0,0 +1,125 @@
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Gabriel Roldan, Boundless Spatial Inc, Copyright 2015
```
This is a new file, so I expect you are the author copyright 2020?

jodygarnett commented on geowebcache/swiftblob/src/main/java/org/geowebcache/swift/SwiftBlobStoreConfigProvider.java:
```diff
@@ -0,0 +1,125 @@
+import org.geowebcache.config.Info;
+import org.geowebcache.config.XMLConfigurationProvider;
+
+public class SwiftBlobStoreConfigProvider implements XMLConfigurationProvider {
```
You may want a public javadoc

jodygarnett commented on geowebcache/swiftblob/src/main/java/org/geowebcache/swift/TMSKeyBuilder.java:
```diff
@@ -0,0 +1,238 @@
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Gabriel Roldan, Boundless Spatial Inc, Copyright 2015
```
as above

dhlambert commented:
@jodygarnett I have updated javadoc and added some documentation in  the configuration/storage.rst file for Openstack Swift Blobstore.  Have also made some refactoring changes within config for some unused properties.

I have not updated the author as the TMSKeyBuilder class is unmodified apart from javadoc, copied from s3storage - see #830. Also the SwiftBlobStoreConfigProvider.java file has some untouched code and some added code. Not sure if I should be taking all the credit for another persons work.

Commits have been left unsquashed for reviewing.

dhlambert commented:
Hi @jodygarnett,

Just wondering if you think this pull request needs any further changes? Also, if there is any proposed timeline for merging?

jodygarnett commented:
> Hi @jodygarnett,
> 
> Just wondering if you think this pull request needs any further changes? Also, if there is any proposed timeline for merging?

Let me check, and yes I am happy to merge.

Please note a release candidate went out last week, after the release goes out we can backport this work also.

jodygarnett commented on documentation/en/user/source/configuration/storage.rst:
```diff
@@ -791,3 +791,63 @@ In this section are described other available configuration parameters to config
 		.. note:: A value of *max-size* bigger or equal to Integer.MAX_VALUE cannot be used in order to avoid an uncontrollable growth of the cache size.
 
+OpenStack Swift (Swift) Blob Store
++++++++++++++++++++++++++++++++++++++++++++++
```
Thanks this is perfect

jodygarnett commented on geowebcache/swiftblob/pom.xml:
```diff
@@ -0,0 +1,57 @@
+        <dependency>
+            <groupId>org.apache.jclouds.api</groupId>
+            <artifactId>openstack-swift</artifactId>
+            <version>${jclouds.version}</version>
```
We try and add version numbers to the root pom.xml dependency management section; something to consider for later.

jodygarnett commented on geowebcache/swiftblob/pom.xml:
```diff
@@ -0,0 +1,57 @@
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.2.4</version>
```
Please use dependency management section to manage version numbers (so we do not end up with conflicts). 
This is a test dependency so you will not get a conflict in this case.

jodygarnett commented on geowebcache/swiftblob/pom.xml:
```diff
@@ -0,0 +1,57 @@
+    </dependencies>
+
+    <build>
+        <plugins>
```
Q: Why empty plugins section?
